test/common/testlib.py

```diff
@@ -1829,6 +1829,10 @@ def write_file(self, path, content, append=False, owner=None, perm=None, post_re
         '''
         m = self.machine
         self.restore_file(path, post_restore_action=post_restore_action)
+        dirname = os.path.dirname(path)
+        exists = m.execute(f"if test -d {dirname}; then echo yes; fi").strip() != ""
+        if not exists:
+            m.execute(f"mkdir -p {dirname}")
```

Comment:
`mkdir -p` already DTRT, you can drop the `test -d`. Thanks for that change, with that we can remove a lot of `m.execute("mkdir ...")` from tests.

Please change the commit message to "create directory", as the path includes the file name.